anyway/user_functions.py

```diff
@@ -21,3 +24,17 @@ def get_current_user() -> Optional[Users]:
     if current_user.is_anonymous:
         return None
     return current_user
+
+
+def is_current_user_have_permission(db: SQLAlchemy, request: Request) -> bool:
+    allowed_roles = db.session.query(RolesToAPI).filter(RolesToAPI.api_name == request.path).all()
```

Comment:
I think reading the roles table into memory when application starts, and working with the memory (refreshing periodically) will result in simpler code.